packages/react-core/src/filters/Filter.js

```diff
@@ -51,7 +16,11 @@ function passesFilter(columns, filters, feature) {
         throw new Error(`"${filter}" filter is not implemented.`);
       }
 
-      return filterFunction(columnFilters[filter].values, feature[column]);
+      return filterFunction(
```

Comment:
I am worried about performance of this search, where filter `stringSearch` function creates regexp for _every_ separate feature.

This flow should be somehow reversed, so all "filter functions" are have change to init before actually processing all features.

Proposal, change signature of filter functions from `Record<string, FilterFunction>` to `Record<string, (fiterValues, params) => (value) => boolean)>`

IOW, filterFunction should contain factories of filters and they should be called before loop.

Reply:
@zbigg thanks for this very interesting comment. What you comment isn't small change, it requires changes in many parts of the app and it isn't the goal of this PR. But, I encourage you to open a RFC for further discussion about a possible refactor of filters.

About your performance worries:
I've been testing the performance of my code and I made some changes to improve it:

![image](https://user-images.githubusercontent.com/9151432/154450512-b0710777-79f4-4c82-b80e-f8474c2e45be.png)

In this _dirty_ benchmark, I execute the code to create the filterValue regexp 1M times.
My original code execution time was nearly 400ms.
Avoiding recreating those static regexp (specialRegExp and normalizedRegExp), avoiding to always execute normalize (this can reduce execution time by half, but it's enabled by default) and using toLowerCase instead of toLocaleLowerCase, the execution time is ~140ms (without normalize, it can be ~70ms).

140ms isn't perfect, and I'm sure that rethinking how the filters are made these numbers would be better, but I think that maybe it's the best we can achieve with what we have right now.

Reply:
Good improvements on static regexps.

> it requires changes in many parts of the app and it
I would argue that this is not big or breaking change. `filterFunctions` is used twice (`passesFilter` and `passesFilterUsingBinary`) in very similar way iterate over filter columns, get filter function and with well proper helper

>  isn't the goal of this PR

If we add new feature and it requires (slight IMO) internal architecture change, it's good place to introduce it.

Reply:
Why does it require it? Your proposal is good and yes, maybe there are ways to achieve it smaller that I thought, but from my knowledge contributing in this lib, I should keep the PRs as simple and as focused in the task as possible. In this PR I already added more that the strictly necessary adding new tests for filters. I don't feel that adding also that is a good idea, because the current proposal doesn't add a performance problem (IMO) and your proposal is an improvement that have to be done in its own task: if we have a problem with stringSearch, we should revert it without affecting your idea and the same about your proposal. Also, when making your idea, we can make changes in other filters, not only stringSearch.

In this case, let @VictorVelarde decides what to do here 😆  I don't think we will reach an agreement.

Reply:
@Clebal We have talked about this and we think that this can be landed as it is, but it should be improved in the future (especially if you find, like Z suspects, performance issues in your UI). 

Please Sergio, give us feedback about your use of this in templates, once you integrate it

Reply:
https://user-images.githubusercontent.com/9151432/154667811-67495003-fe79-4662-9832-d5849dc28cdd.mov


I've done some tests in Site Selection, where I need this filter and it's works really great. The search has a minor debounce of 100ms, but the filter process is almost instant.

Reply:
I've been testing it in Site Selection and it's quite instant with 23K features, filtering in worker and in layer.